tests/common/macros.rs

```diff
@@ -46,3 +46,38 @@ macro_rules! path_concat {
         pb.to_str().unwrap().to_owned()
     }};
 }
+
+#[macro_export]
+macro_rules! utility_test {
+    () => (
+        fn util_name<'a>() -> &'a str {
+            let split : Vec<&str> = module_path!().split("_").collect();
+            split[1]
```

Comment:
>```module_path!().split("_").nth(1).expect("no test name")```

added

> how about using macros? I think that's less surprising. i.e.

I'm opposed to this. I think that exclamation point is a readability eyesore to have everywhere. I'd actually prefer to have the helper functions in every test file than to have a macro call in every test case.